e2e/framework/machineset.go

```diff
@@ -19,6 +19,7 @@ type machineSetParams struct {
 	failureDomain     string
 	replicas          int32
 	infrastructureRef corev1.ObjectReference
+	UserDataSecret    string
```

Comment:
This should probably have been unexported like the rest of the fields in this struct. But we can fix it in a followup PR I guess, if we are in a rush.

Reply:
Agreed; I wanted to avoid changing every call site.  Happy to fix it in a follow-up.

Reply:
There are, including this PR, around 5 call sites, I'd prefer we just get that updated, rather than putting this off for the future